### PR TITLE
-updated view other member's health records

### PR DESCRIPTION
**Pull request by amit-metri**
-updated view other member's health records: HAPP-413
-bug fix: HAPP-607

### CODE REVIEW

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/HealthRecordsFragment.kt [hunk 1]
```diff
@@ -72,25 +68,35 @@ class HealthRecordsFragment : Fragment(R.layout.fragment_health_records) {
...
-                        spanSizeLookup =
-                            object : GridLayoutManager.SpanSizeLookup() {
-                                override fun getSpanSize(position: Int) = when (position) {
-                                    0 -> GRID_SPAN_COUNT
-                                    else -> LIST_SPAN_COUNT
-                                }
-                            }
-                    }
+                if (records.size == 1) {
+                    val navOptions = NavOptions.Builder()
```
Comment (rashmi-bambhania) [inline app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/HealthRecordsFragment.kt:83]: After adding any covid or vaccine public record successfully and when you come back to this screen does it update newly added record folder ? Please test this scenario.

File: app/src/main/java/ca/bc/gov/bchealth/ui/setting/ProfileFragment.kt [hunk 2]
```diff
@@ -85,8 +86,9 @@ class ProfileFragment : Fragment(R.layout.fragment_profile) {
                         return@collect
                     } else {
                         initClickListeners()
-                        binding.layoutProfile.isVisible = it.isLoggedIn
-                        binding.layoutLogin.isVisible = !it.isLoggedIn
+                        val loginStatus = it.loginSessionStatus == LoginSessionStatus.ACTIVE
```
Comment (rashmi-bambhania) [inline app/src/main/java/ca/bc/gov/bchealth/ui/setting/ProfileFragment.kt:89]: As you are checking that login session is active or expired, I suggest to rename loginStatus variable to isLoginStatusActive or any other suitable variable name which increases readability.

File: app/src/main/java/ca/bc/gov/bchealth/ui/setting/SettingFragment.kt [hunk 3]
```diff
@@ -98,8 +99,9 @@ class SettingFragment : Fragment(R.layout.fragment_setting) {
 
                     showLoader(it.showLoading)
 
-                    binding.switchLogin.isChecked = it.isLoggedIn
-                    isLoggedIn = it.isLoggedIn
+                    val loginStatus = it.loginSessionStatus == LoginSessionStatus.ACTIVE
```
Comment (rashmi-bambhania) [inline app/src/main/java/ca/bc/gov/bchealth/ui/setting/SettingFragment.kt:102]: Here also update loginStatus variable name

**Review by rashmi-bambhania — CHANGES_REQUESTED** [review]
Added few comments

File: app/src/main/java/ca/bc/gov/bchealth/ui/setting/SettingFragment.kt (hunk 3 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/bchealth/ui/setting/SettingFragment.kt:102]: taken care.

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/HealthRecordsFragment.kt (hunk 1 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/HealthRecordsFragment.kt:83]: this will be taken care

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**